x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/InferenceEndpointKey.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.queries;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.inference.MinimalServiceSettings;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class InferenceEndpointKey implements Writeable {
```

Comment:
I like this conceptually, but nitpicky - I would like to find a better name for it.